---
title: "Figure S9"
author: Beata Szeitz
output:
  html_notebook:
    toc: yes
---

```{r Load packages}
source("Install_load_packages.R")
source("Utility_functions.R")
source("Color_list.R")
```
 
```{r Load files}
Annot <- read_in_with_rowname("Files/Annotation",F)
row.names(Annot) <- Annot$Sample
CP.expr.imp <- read_in_with_rowname("Files/CP_expression_filtered_imputed",F)[,Annot$Sample]
CM.expr.imp <- read_in_with_rowname("Files/CM_expression_filtered_imputed",F)[,Annot$Sample]
Secreted.proteins <- read_in_with_rowname("Files/Tier1_and_2_secreted",F)[,1]
CM.expr.imp.secr <- CM.expr.imp[row.names(CM.expr.imp) %in% Secreted.proteins,]
prot.to.gene <- read_in_with_rowname("Files/Protein_to_Gene",F)
row.names(prot.to.gene) <- prot.to.gene$Protein
splsda.res <- read_in_with_rowname("Files/sPLSDA_results_extended",F)
splsda.res$Accession <- row.names(splsda.res)
DEA.subtypes.CCLE <- read_in_with_rowname("Files/DEA_Subtypes_CCLE",F)
DEA.subtypes.CP <- read_in_with_rowname("Files/CP_DEA_Subtype",F)
DEA.subtypes.CM <- read_in_with_rowname("Files/CM_DEA_Subtype",F)
CCLE.normalized <- read_in_with_rowname("Files/CCLE_expression_normalized",F) 
CCLE.info <- read_in_with_rowname("Files/CCLE_rowname_to_geneID",F)
George.Zscore <- read_in_with_rowname("Files/George_expression_Zscores",F)
Annot.CCLE <- read_in_with_rowname("Files/CCLE_annotation",F)
Annot.CCLE$Sample <- Annot.CCLE$CCLE_ID
Annot.CCLE$Subtype <- Annot.CCLE$`Subtype assignment`
Annot.George <- read_in_with_rowname("Files/George_annotation",F)
colnames(Annot.George)[2] <- "Subtype"
row.names(Annot.George) <- Annot.George$Sample

markers <- c(splsda.res$Gene, "ASCL1", "NEUROD1", "YAP1")

George.FPKM <- read_in_with_rowname("Files/George_expression_FPKM",F)
all(row.names(George.FPKM) == row.names(George.Zscore))
George.genes <- data.frame(Gene = unlist(lapply(row.names(George.FPKM), function(x){
  strsplit(x, split="_")[[1]][1]
})),
Row.name = row.names(George.FPKM))

#remove dupl. genes (keep only the transcript with highest rowsum)
dupl.genes <- unique(George.genes[duplicated(George.genes$Gene),"Gene"])

to.delete <- vector()

for (i in 1:length(dupl.genes)){
  transcripts <- George.genes[George.genes$Gene == dupl.genes[i],"Row.name"]
  rowsums <- apply(George.FPKM[transcripts,], 1, function(x){
    sum(x, na.rm = T)
  })
  names(rowsums) <- transcripts
  rowsums <- rowsums[order(rowsums, decreasing = T)]
  to.delete <- c(to.delete, names(rowsums[-1]))
}

George.genes <- George.genes[!George.genes$Row.name %in% to.delete,]
George.genes.markers <- George.genes[George.genes$Gene %in% markers,]
George.FPKM.markers <- George.FPKM[George.genes.markers$Row.name,]
George.Zscore.markers <- George.Zscore[George.genes.markers$Row.name,]

```



```{r }
#DEA.subtypes.George <- Kruskal_posthoc_adj(George.Zscore, Annot.George, "Subtype",
#                                         c("SCLC-A", "SCLC-N","SCLC-P", "SCLC-Y"), analysisName="")
#export_with_rowname(DEA.subtypes.George, "Files/DEA_Subtypes_George", T)
DEA.subtypes.George <- read_in_with_rowname("Files/DEA_Subtypes_George",F)
DEA.subtypes.George$Gene <- unlist(lapply(DEA.subtypes.George$Accession, function(x){
  strsplit(x, split="_")[[1]][1]
}))

DEA.subtypes.George <- DEA.subtypes.George[DEA.subtypes.George$Accession %in% row.names(George.Zscore.markers),]
```



```{r Baseline expression}
SA.samples <- Annot.George[Annot.George$Subtype=="SCLC-A","Sample"]
SN.samples <- Annot.George[Annot.George$Subtype=="SCLC-N","Sample"]
SP.samples <- Annot.George[Annot.George$Subtype=="SCLC-P","Sample"]
SY.samples <- Annot.George[Annot.George$Subtype=="SCLC-Y","Sample"]

Mean.Expression <- data.frame(Accession = row.names(George.FPKM.markers),
                              `Mean.SCLC-A` = NA,
                              `Mean.SCLC-N` = NA,
                              `Mean.SCLC-P` = NA,
                              `Mean.SCLC-Y` = NA,
                              `Max.SCLC-A` = NA,
                              `Max.SCLC-N` = NA,
                              `Max.SCLC-P` = NA,
                              `Max.SCLC-Y` = NA,
                              `Min.SCLC-A` = NA,
                              `Min.SCLC-N` = NA,
                              `Min.SCLC-P` = NA,
                              `Min.SCLC-Y` = NA,
                              `Mean.SCLC-A.sc` = NA,
                              `Mean.SCLC-N.sc` = NA,
                              `Mean.SCLC-P.sc` = NA,
                              `Mean.SCLC-Y.sc` = NA,
                              `Max.SCLC-A.sc` = NA,
                              `Max.SCLC-N.sc` = NA,
                              `Max.SCLC-P.sc` = NA,
                              `Max.SCLC-Y.sc` = NA,
                              `Min.SCLC-A.sc` = NA,
                              `Min.SCLC-N.sc` = NA,
                              `Min.SCLC-P.sc` = NA,
                              `Min.SCLC-Y.sc` = NA)

for (i in 1:nrow(George.FPKM.markers)){
  Mean.Expression$Mean.SCLC.A[i] <- mean(as.numeric(George.FPKM.markers[i,SA.samples]), na.rm=T)
  Mean.Expression$Mean.SCLC.N[i] <- mean(as.numeric(George.FPKM.markers[i,SN.samples]), na.rm=T)
  Mean.Expression$Mean.SCLC.P[i] <- mean(as.numeric(George.FPKM.markers[i,SP.samples]), na.rm=T)
  Mean.Expression$Mean.SCLC.Y[i] <- mean(as.numeric(George.FPKM.markers[i,SY.samples]), na.rm=T)
  
  Mean.Expression$Max.SCLC.A[i] <- max(as.numeric(George.FPKM.markers[i,SA.samples]), na.rm=T)
  Mean.Expression$Max.SCLC.N[i] <- max(as.numeric(George.FPKM.markers[i,SN.samples]), na.rm=T)
  Mean.Expression$Max.SCLC.P[i] <- max(as.numeric(George.FPKM.markers[i,SP.samples]), na.rm=T)
  Mean.Expression$Max.SCLC.Y[i] <- max(as.numeric(George.FPKM.markers[i,SY.samples]), na.rm=T)
  
  Mean.Expression$Min.SCLC.A[i] <- min(as.numeric(George.FPKM.markers[i,SA.samples]), na.rm=T)
  Mean.Expression$Min.SCLC.N[i] <- min(as.numeric(George.FPKM.markers[i,SN.samples]), na.rm=T)
  Mean.Expression$Min.SCLC.P[i] <- min(as.numeric(George.FPKM.markers[i,SP.samples]), na.rm=T)
  Mean.Expression$Min.SCLC.Y[i] <- min(as.numeric(George.FPKM.markers[i,SY.samples]), na.rm=T)
  
  
  Mean.Expression$Mean.SCLC.A.sc[i] <- mean(as.numeric(George.Zscore.markers[i,SA.samples]), na.rm=T)
  Mean.Expression$Mean.SCLC.N.sc[i] <- mean(as.numeric(George.Zscore.markers[i,SN.samples]), na.rm=T)
  Mean.Expression$Mean.SCLC.P.sc[i] <- mean(as.numeric(George.Zscore.markers[i,SP.samples]), na.rm=T)
  Mean.Expression$Mean.SCLC.Y.sc[i] <- mean(as.numeric(George.Zscore.markers[i,SY.samples]), na.rm=T)
  
  Mean.Expression$Max.SCLC.A.sc[i] <- max(as.numeric(George.Zscore.markers[i,SA.samples]), na.rm=T)
  Mean.Expression$Max.SCLC.N.sc[i] <- max(as.numeric(George.Zscore.markers[i,SN.samples]), na.rm=T)
  Mean.Expression$Max.SCLC.P.sc[i] <- max(as.numeric(George.Zscore.markers[i,SP.samples]), na.rm=T)
  Mean.Expression$Max.SCLC.Y.sc[i] <- max(as.numeric(George.Zscore.markers[i,SY.samples]), na.rm=T)
  
  Mean.Expression$Min.SCLC.A.sc[i] <- min(as.numeric(George.Zscore.markers[i,SA.samples]), na.rm=T)
  Mean.Expression$Min.SCLC.N.sc[i] <- min(as.numeric(George.Zscore.markers[i,SN.samples]), na.rm=T)
  Mean.Expression$Min.SCLC.P.sc[i] <- min(as.numeric(George.Zscore.markers[i,SP.samples]), na.rm=T)
  Mean.Expression$Min.SCLC.Y.sc[i] <- min(as.numeric(George.Zscore.markers[i,SY.samples]), na.rm=T)
}


Mean.Expression$Gene <- unlist(lapply(Mean.Expression$Accession, function(x){
  strsplit(x, split="_")[[1]][1]
}))

DEA.subtypes.George.2 <- merge(Mean.Expression, DEA.subtypes.George, by="Accession")
DEA.subtypes.George.2$Gene <- unlist(lapply(DEA.subtypes.George.2$Accession, function(x){
  strsplit(x, split="_")[[1]][1]
}))

```







```{r }
DEA.subtypes.proteomics <- rbind(DEA.subtypes.CP[splsda.res[splsda.res$Data=="CP","Accession"],],
                                 DEA.subtypes.CM[splsda.res[splsda.res$Data=="CM","Accession"],],
                                 DEA.subtypes.CP[c("Q13562", "P46937", "P50553"),])
row.names(DEA.subtypes.proteomics)[nrow(DEA.subtypes.proteomics)] <- "P50553"
DEA.subtypes.proteomics$Gene <- prot.to.gene[row.names(DEA.subtypes.proteomics),"Gene"]


Marker.table <- merge(splsda.res, DEA.subtypes.proteomics, by="Gene", all = T)

row.names(DEA.subtypes.CCLE) <- DEA.subtypes.CCLE$Gene
Marker.table <- merge(Marker.table, DEA.subtypes.CCLE[markers,], by="Gene", all = T)
Marker.table <- subset(Marker.table, !is.na(Marker.table$Gene))

Marker.table <- merge(Marker.table, DEA.subtypes.George.2, by="Gene", all = T)

Marker.table$Subtype <- ifelse(Marker.table$Gene =="ASCL1", "UP in A", Marker.table$Subtype)
Marker.table$Subtype <- ifelse(Marker.table$Gene =="YAP1" | Marker.table$Gene =="YAP1 ", "UP in Y", Marker.table$Subtype)
Marker.table$Subtype <- ifelse(Marker.table$Gene =="NEUROD1", "UP in N", Marker.table$Subtype)

wb <- createWorkbook(
  title = NULL,
)
sheetnames <- c("UP in AvsN","UP in NvsA","UP in P", "DOWN in P", "UP in Y", "DOWN in Y", 'All')
sheets <- list(Marker.table[Marker.table$Subtype == "UP in AvsN",],
               Marker.table[Marker.table$Subtype == "UP in NvsA",],
               Marker.table[Marker.table$Subtype == "UP in P" ,],
               Marker.table[Marker.table$Subtype == "DOWN in P",],
               Marker.table[Marker.table$Subtype == "UP in Y" ,],
               Marker.table[Marker.table$Subtype == "DOWN in Y" ,],
               Marker.table)

for (i in 1:length(sheets)) {
  addWorksheet(wb, sheetnames[i])
  writeData(
    wb = wb,
    sheet = sheetnames[i],
    x = sheets[[i]])
}
#saveWorkbook(wb, paste0("Files/SCLC_sPLSDA_markers_in_transcriptomics.xlsx"), overwrite = TRUE)


```


```{r }
Markers.list <- list(Marker.table[Marker.table$Subtype == "UP in AvsN",],
               Marker.table[Marker.table$Subtype == "UP in NvsA",],
               Marker.table[Marker.table$Subtype == "UP in P" ,],
               Marker.table[Marker.table$Subtype == "DOWN in P",],
               Marker.table[Marker.table$Subtype == "UP in Y" ,],
               Marker.table[Marker.table$Subtype == "DOWN in Y" ,])

Markers.list[[1]] <- subset(Markers.list[[1]], Markers.list[[1]]$`P.Value(SCLC-N-SCLC-A)` < 0.05 |
                              Markers.list[[1]]$`wilcox(SCLC-N-SCLC-A)_` < 0.05)

Markers.list[[2]] <- subset(Markers.list[[2]], Markers.list[[2]]$`P.Value(SCLC-N-SCLC-A)` < 0.05 |
                              Markers.list[[2]]$`wilcox(SCLC-N-SCLC-A)_` < 0.05)

Markers.list[[3]] <- subset(Markers.list[[3]], 
                            Markers.list[[3]]$`P.Value(SCLC-P-SCLC-A)` < 0.05 |
                               Markers.list[[3]]$`P.Value(SCLC-P-SCLC-N)` < 0.05 |
                               Markers.list[[3]]$`P.Value(SCLC-Y-SCLC-P)` < 0.05 |
                              Markers.list[[3]]$`wilcox(SCLC-P-SCLC-A)_` < 0.05 |
                              Markers.list[[3]]$`wilcox(SCLC-P-SCLC-N)_` < 0.05 |
                              Markers.list[[3]]$`wilcox(SCLC-Y-SCLC-P)_` < 0.05)

Markers.list[[4]] <- subset(Markers.list[[4]], 
                            Markers.list[[4]]$`P.Value(SCLC-P-SCLC-A)` < 0.05 |
                               Markers.list[[4]]$`P.Value(SCLC-P-SCLC-N)` < 0.05 |
                               Markers.list[[4]]$`P.Value(SCLC-Y-SCLC-P)` < 0.05 |
                              Markers.list[[4]]$`wilcox(SCLC-P-SCLC-A)_` < 0.05 |
                              Markers.list[[4]]$`wilcox(SCLC-P-SCLC-N)_` < 0.05 |
                              Markers.list[[4]]$`wilcox(SCLC-Y-SCLC-P)_` < 0.05)

Markers.list[[5]] <- subset(Markers.list[[5]], 
                            Markers.list[[5]]$`P.Value(SCLC-Y-SCLC-A)` < 0.05 |
                               Markers.list[[5]]$`P.Value(SCLC-Y-SCLC-N)` < 0.05 |
                               Markers.list[[5]]$`P.Value(SCLC-Y-SCLC-P)` < 0.05 |
                              Markers.list[[5]]$`wilcox(SCLC-Y-SCLC-A)_` < 0.05 |
                              Markers.list[[5]]$`wilcox(SCLC-Y-SCLC-N)_` < 0.05 |
                              Markers.list[[5]]$`wilcox(SCLC-Y-SCLC-P)_` < 0.05)

Markers.list[[6]] <- subset(Markers.list[[6]], 
                            Markers.list[[6]]$`P.Value(SCLC-Y-SCLC-A)` < 0.05 |
                               Markers.list[[6]]$`P.Value(SCLC-Y-SCLC-N)` < 0.05 |
                               Markers.list[[6]]$`P.Value(SCLC-Y-SCLC-P)` < 0.05 |
                              Markers.list[[6]]$`wilcox(SCLC-Y-SCLC-A)_` < 0.05 |
                              Markers.list[[6]]$`wilcox(SCLC-Y-SCLC-N)_` < 0.05 |
                              Markers.list[[6]]$`wilcox(SCLC-Y-SCLC-P)_` < 0.05)


```




```{r }

selected.markers <- na.omit(unique(c(Markers.list[[1]]$Protein, Markers.list[[2]]$Protein,
                           Markers.list[[3]]$Protein, Markers.list[[4]]$Protein,
                           Markers.list[[5]]$Protein, Markers.list[[6]]$Protein)))
selected.markers <- selected.markers


selected.markers.George <- unique(na.omit(c(Markers.list[[1]]$Accession.y, Markers.list[[2]]$Accession.y,
                           Markers.list[[3]]$Accession.y, Markers.list[[4]]$Accession.y,
                           Markers.list[[5]]$Accession.y, Markers.list[[6]]$Accession.y)))

selected.markers.George <- selected.markers.George[!grepl("YAP1", selected.markers.George) &
                                                 !grepl("ASCL1", selected.markers.George) &
                                                 !grepl("NEUROD1", selected.markers.George)]

selected.markers.CCLE <- unique(na.omit(c(Markers.list[[1]]$Gene, Markers.list[[2]]$Gene,
                           Markers.list[[3]]$Gene, Markers.list[[4]]$Gene,
                           Markers.list[[5]]$Gene, Markers.list[[6]]$Gene)))

selected.markers.CCLE <- selected.markers.CCLE[!grepl("YAP1", selected.markers.CCLE) &
                                                 !grepl("ASCL1", selected.markers.CCLE) &
                                                 !grepl("NEUROD1", selected.markers.CCLE)]

George.Zscore.melt <- George.Zscore[selected.markers.George,]
George.Zscore.melt$Var1 <- row.names(George.Zscore.melt)
George.Zscore.melt <- melt(George.Zscore.melt)
colnames(George.Zscore.melt) <- c("Transcript", "Sample", "Z.score")
George.Zscore.melt$Gene <-unlist(lapply(George.Zscore.melt$Transcript, function(x){
  strsplit(x, split="_")[[1]][1]
}))
George.Zscore.melt$Data <- "George et al."
George.Zscore.melt <- merge(George.Zscore.melt, Annot.George[,c("Sample", "Subtype")], by="Sample")


selected.markers.CCLE.transcript <- CCLE.info[CCLE.info$hgnc_symbol %in% selected.markers.CCLE,"Row.name"]
CCLE.normalized.melt <- melt(t(scale(t(CCLE.normalized[selected.markers.CCLE.transcript,]))))
colnames(CCLE.normalized.melt) <- c("Transcript", "Sample", "Z.score")
CCLE.normalized.melt$Gene <-unlist(lapply(CCLE.normalized.melt$Transcript, function(x){
  CCLE.info[CCLE.info$Row.name==x,"hgnc_symbol"]
}))
CCLE.normalized.melt$Data <- "CCLE"
CCLE.normalized.melt <- merge(CCLE.normalized.melt, Annot.CCLE[,c("Sample", "Subtype")], by="Sample")



Selected.Marker.Expr <- rbind(George.Zscore.melt, CCLE.normalized.melt)
Selected.Marker.Expr$Transcript <- NULL

create_decoy <- function(genename, data.missing){
  data.frame(Sample = c("X","Y","Z","A"),
             Z.score = c(0,0,0,0),
             Gene = c(genename, genename, genename,genename),
             Data=data.missing,
             Subtype = c("SCLC-A","SCLC-N","SCLC-P","SCLC-Y"))
}

Selected.Marker.Expr.ext <- rbind(Selected.Marker.Expr, 
                         create_decoy("CAVIN1","George et al."),
                         create_decoy("CYBC1","George et al."),
                         create_decoy("HNRNPLL","George et al."),
                         create_decoy("SINHCAF","George et al."),
                         create_decoy("ERO1B","George et al."),
                         create_decoy("METTL26","George et al."),
                         create_decoy("RIOX2","George et al."),
                         create_decoy("LRMP","CCLE"),
                         create_decoy("TMEM106B","CCLE"),
                         create_decoy("TMEM189","CCLE"),
                         create_decoy("FAM49A","CCLE"))



Selected.Marker.Expr.ext <- Selected.Marker.Expr.ext[Selected.Marker.Expr.ext$Gene %in% selected.markers.CCLE,]

Selected.Marker.Expr.ext$Subtype <- unlist(lapply(Selected.Marker.Expr.ext$Subtype, function(x){
  strsplit(x, split="-")[[1]][2]
}))

```
```{r }
Marker.table$Pattern <- Marker.table$Subtype
Selected.Marker.Expr.ext <- merge(Selected.Marker.Expr.ext, Marker.table[,c("Gene","Pattern" )], by="Gene")

```



```{r , fig.width=20, fig.height=30}
my_comparisons <- list( c("N", "A"),c("P", "A"), c("N", "P"), c("Y", "A"), c("Y", "N"), c("Y", "P") )

selected.markers.CCLE.significant <- selected.markers.CCLE[!selected.markers.CCLE %in% 
                                                             c("MAP2K6","SHANK2",
                                                               "GPN1", "GYBC1",
                                                               "EIF2B4", "ARHGAP32",
                                                               "AP2M1", "DNAJC7", "IL18")]

glist <- list()


for (i in 1:length(selected.markers.CCLE.significant)){

  Selected.Marker.Expr.ext.sub <- Selected.Marker.Expr.ext[Selected.Marker.Expr.ext$Gene == selected.markers.CCLE.significant[i],]
 g <- ggplot(Selected.Marker.Expr.ext.sub, aes(x=Subtype, y=Z.score, fill=Subtype))+ theme_bw()+# +geom_violin()  
    scale_color_manual(values=c("#0000FE", "#21FE06","#FEFE0A", "#FB02FE"))+ 
   scale_fill_manual(values=c("#0000FE", "#21FE06","#FEFE0A", "#FB02FE"))+ 
   geom_boxplot() + ylab("Z-score")+xlab("")+
   ggtitle(paste0(selected.markers.CCLE.significant[i], " - ",Selected.Marker.Expr.ext.sub[1,"Pattern"]))+
          #stat_compare_means(label = "p.signif", method = "anova")+ theme_bw() + #label.y = labelpos, 
          #stat_compare_means(comparisons = my_comparisons,label = "p.signif", method = "t.test")+
   theme(legend.position="none")
 #print(g+ facet_wrap(Data ~ ., ncol=3, nrow=1))
 glist[[i]] <- g+ facet_wrap(Data ~ ., ncol=2, nrow=1)
}

glist2 <- glist[sapply(glist, function(x) length(x) > 3)]

g <- grid.arrange(grobs=glist2, ncol=7)
g
ggsave(file="Figures/Markers_expression_in_transcriptomics_significant.pdf", g)
ggsave(file="Figures/Markers_expression_in_transcriptomics_significant.svg", g)

```
